iceoryx_posh/test/moduletests/test_popo_client_options.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_posh/popo/client_options.hpp"
+
+#include "test.hpp"
+
+namespace
+{
+using namespace ::testing;
+
+TEST(ClientOptions_test, SerializationRoundTripIsSuccessful)
```

Comment:
Please add a test with:
- deserializing an completely invalid serialization (maybe from one int or so)
- deserializing where the two policies have invalid integer values (to check if the default is set)